astropy/io/fits/hdu/image.py

```diff
@@ -135,9 +147,31 @@ def __init__(self, data=None, header=None, do_not_scale_image_data=False,
                 self._data_needs_rescale = True
             return
         else:
+            # Setting data will set set _bitpix, _bzero, and _bscale to the
+            # appropriate BITPIX for the data, and always sets _bzero=0 and
+            # _bscale=1.
             self.data = data
             self.update_header()
```

Comment:
Indeed, passes locally, will fix in separate PR addressing the remaining comments too.